source/aws_signing.c

```diff
@@ -1715,6 +1725,45 @@ static int s_build_canonical_request_body_chunk(struct aws_signing_state_aws *st
     return AWS_OP_SUCCESS;
 }
 
+static int s_build_canonical_request_trailing_headers(struct aws_signing_state_aws *state) {
+
```

Comment:
Ok, so we shouldn't have followed the chunk signing pattern here.  Like request signing, this should get written to the canonical_request member, and then the hex-encoded sha256 of the "canonical request" should get written to the state via s_build_canonical_request_hash.  Obviously, drop all of the string-to-sign components out of this function.

Then in the string-to-sign construction for this signing type, we should use that value appropriately when we build the trailing header string-to-sign